### PR TITLE
[#1183] Fix internal server error when sending malformed authentication header

### PR DESCRIPTION
**Pull request by kaniyan**
Fix for issue #1183 by using a `RoutingContextDecorator`.

### CODE REVIEW

File: service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java [hunk 1]
```diff
@@ -70,7 +72,14 @@ protected void processException(final RoutingContext ctx, final Throwable except
 
     @Override
     public void parseCredentials(final RoutingContext context, final Handler<AsyncResult<JsonObject>> handler) {
-        super.parseCredentials(context, ar -> {
+        final RoutingContextDecorator routingContextDecorator = new RoutingContextDecorator(context.currentRoute(),
+                context) {
+
+            public void fail(final Throwable throwable) {
+                context.fail(HttpURLConnection.HTTP_UNAUTHORIZED, throwable);
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java]: wouldn't be sufficient to add this line in an `else` block in the handler below? Using the `RoutingContextDecorator` should probably work but I think we can achieve the same result with a little less *ammo*, don't you think?
Can you also add a corresponding unit test?

Comment (kaniyan) [reply service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java]: @sophokles73 Already I checked for the `else` block, it doesn't work. Vertx `BasicAuthHandlerImpl` invokes `context.fail(e)` when error occurs, thereby the `DefaultFailureHandler` is being invoked.

Comment (sophokles73) [reply service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java]: I see. Can you then please add a comment explaining why using the `RoutingContextDecorator` in this case is necessary?

Comment (kaniyan) [reply service-base/src/main/java/org/eclipse/hono/service/http/HonoBasicAuthHandler.java]: @sophokles73 I have added a unit test and a comment for the above.

**Pull request merged**